lib/trln_argon/view_helpers/syndetics_helper.rb

```diff
@@ -32,6 +32,10 @@ def get_params(doc, client = 'trlnet', format_spec = 'XML.XML')
           oclc: doc.fetch(TrlnArgon::Fields::OCLC_NUMBER, []).dup
           # placeholder for UPC and EAN when they are added
         }
+        # add upc if available
+        if doc.fetch(TrlnArgon::Fields::UPC, []).present?
```

Comment:
I think I'd prefer if there were a new method on SolrDocument (with a test) to deal with munging the UPC. Something like:
```
#lib/trln_argon/solr_document.rb
def upc
      fetch(TrlnArgon::Fields::UPC, []).map do |upc|
        upc.to_s.gsub(/\D/, '')
      end.reject(&:empty?)
end
```

There's also text other than "UPC: " present in the field so it's probably better to remove any non-digits from the field (`/\D/` instead of `'UPC: '`).

If you just want the first UPC (there can be more than one) then I'd write the syndetics helper like:

```
        params = {
          isbn: [doc.fetch(TrlnArgon::Fields::ISBN_NUMBER, []).first],
          oclc: doc.fetch(TrlnArgon::Fields::OCLC_NUMBER, []).dup,
          upc: [doc.upc.first]
          # placeholder for EAN when they are added
        }
```

Reply:
Thanks Cory - I'll give it a try. I couldn't get any sort of thing to work with the existing get_params code block (which is supposed to remove the empty UPC params) and it would never pass the existing tests, hence the approach I took. 

Would you be Ok with a hybrid of your suggestion (defining UPC in solr_document but adding it to the params list independently?

Reply:
I think the tests will pass if you remove empty values from the upc array (`.reject(&:empty?)`)

Reply:
@corylown - should be working fine now. Thanks again for the help!